rootfs/etc/profile.d/dietpi-unsupported_terminal.sh

```diff
@@ -26,10 +26,10 @@ if [[ $SSH_TTY ]] && ! toe -a | grep -q "$TERM"; then
 	fi
 
 	G_WHIP_DEFAULT_ITEM=1
-	G_PROGRAM_NAME='Unsupported SSH client terminal' G_WHIP_MENU "[WARNING] The terminal passed by your SSH client (TERM=$TERM_old) is not supported by this DietPi server.\n
+	
+	(( ! G_PROGRAM_NAME='Unsupported SSH client terminal' G_WHIP_MENU "[WARNING] The terminal passed by your SSH client (TERM=$TERM_old) is not supported by this DietPi server.\n
 To enable usage, we were fooling the server by setting \"TERM=$TERM\". This is not the cleanest solution, since commands might expect colours or formats, that are not supported by the actual terminal.\n
-Please change your SSH clients terminal (passed $TERM string)$ncurses_term."
-	(( ! $? && G_WHIP_RETURNED_VALUE )) && G_AGI ncurses-term
+Please change your SSH clients terminal (passed $TERM string)$ncurses_term." && G_WHIP_RETURNED_VALUE )) && G_AGI ncurses-term
```

Comment:
As above, G_WHIP needs to succeed (no `!`), otherwise an error occurred or the user cancelled the dialog.

This would work:
```
G_PROGRAM_NAME='Unsupported SSH client terminal' G_WHIP_MENU "[WARNING] The terminal passed by your SSH client (TERM=$TERM_old) is not supported by this DietPi server.\n
Please change your SSH clients terminal (passed $TERM string)$ncurses_term." && (( $G_WHIP_RETURNED_VALUE )) && G_AGI ncurses-term
```